cmd/observe/observe_filter_test.go

```diff
@@ -86,20 +88,31 @@ func TestFilterDispatch(t *testing.T) {
 	}))
 
 	require.NoError(t, handleArgs(f))
-
-	assert.Equal(t, []*pb.FlowFilter{
-		{
-			SourceIp:  []string{"1.2.3.4", "5.6.7.8"},
-			Verdict:   []pb.Verdict{pb.Verdict_DROPPED},
-			EventType: []*pb.EventTypeFilter{{Type: 129}},
+	if diff := cmp.Diff(
+		[]*pb.FlowFilter{
+			{
+				SourceIp:  []string{"1.2.3.4", "5.6.7.8"},
+				Verdict:   []pb.Verdict{pb.Verdict_DROPPED},
+				EventType: []*pb.EventTypeFilter{{Type: 129}},
+			},
 		},
-	}, f.whitelist.flowFilters(), "whitelist filter is incorrect")
-
-	assert.Equal(t, []*pb.FlowFilter{
-		{
-			DestinationIp: []string{"5.5.5.5"},
+		f.whitelist.flowFilters(),
+		cmpopts.IgnoreUnexported(pb.FlowFilter{}),
+		cmpopts.IgnoreUnexported(pb.EventTypeFilter{}),
```

Comment:
is this just future-proofing, or the test doesn't pass without these?

Reply:
When omitted, one gets a panic with the following message:
```
--- FAIL: TestFilterDispatch (0.00s)
panic: cannot handle unexported field at {[]*flow.FlowFilter}[0].EventType[0].state:
        "github.com/cilium/cilium/api/v1/flow".EventTypeFilter
consider using a custom Comparer; if you control the implementation of type, you can also consider using an Exporter, AllowUnexported, or cmpopts.IgnoreUnexported [recovered]
        panic: cannot handle unexported field at {[]*flow.FlowFilter}[0].EventType[0].state:
        "github.com/cilium/cilium/api/v1/flow".EventTypeFilter
consider using a custom Comparer; if you control the implementation of type, you can also consider using an Exporter, AllowUnexported, or cmpopts.IgnoreUnexported
```